dap/DapUtils.cc

```diff
@@ -484,64 +520,155 @@ std::string too_big_error_prolog(uint64_t max_response_size_bytes, uint64_t max_
+        }
+        else {
+            // It's a DAP4 thing...
+            msg << "You can find detailed information about DAP4 variable sub-setting here:\n";
+            msg << "https://github.com/OPENDAP/dap4-specification/blob/main/";
+            msg << "01_data-model-and-serialized-rep.md#8-constraints\n";
+        }
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief Throws an exception if the DMR, as marked, would produce a larger than allowed response, or if a requested variable is too big.
+ * The maximum size values are produced by the dap_utils::get_max_sizes_bytes() function.
+ * @param dmr The DMR to which a ce has been applied, or that is otherwise marked for transmission.
+ * @param file The file of the calling code.
+ * @param line The line in the calling code.
+ */
+void throw_if_too_big(libdap::DMR &dmr, const string &file, const unsigned int line){
+    uint64_t max_var_size_bytes=0;
+    uint64_t max_response_size_bytes=0;
+    std::vector<std::string> too_big_vars;
+
+    get_max_sizes_bytes(max_response_size_bytes, max_var_size_bytes);
+    BESDEBUG(MODULE, prolog << "max_var_size_bytes: " << max_var_size_bytes << "\n");
+    BESDEBUG(MODULE, prolog << "max_response_size_bytes: " << max_response_size_bytes << "\n");
+
+    auto response_size_bytes = compute_response_size_and_inv_big_vars(dmr, max_var_size_bytes, too_big_vars);
+    BESDEBUG(MODULE, prolog << "response_size_bytes: " << response_size_bytes << "\n");
+    BESDEBUG(MODULE, prolog << "too_big_vars: " << too_big_vars.size() << "\n");
+
+    stringstream too_big_message;
+    if(its_too_big(
+            too_big_message,
+            max_response_size_bytes,
+            response_size_bytes,
+            max_var_size_bytes,
+            too_big_vars)) {
+        BESDEBUG(MODULE, prolog << "It's TOO BIG:\n" << too_big_message.str() << "\n");
+        throw BESSyntaxUserError(too_big_message.str(), file, line);
+    }
+}
+
+
+/**
+ * @brief Throws an exception if the DDS, as marked, would produce a larger than allowed response, or if a requested variable is too big.
+ * The maximum size values are produced by the dap_utils::get_max_sizes_bytes() function.
+ * @param dds The DDS to which a ce has been applied, or that is otherwise marked for transmission.
+ * @param file The file of the calling code.
+ * @param line The line in the calling code.
+*/
+void throw_if_too_big(const libdap::DDS &dds, const std::string &file, const unsigned int line)
+{
+    uint64_t max_var_size_bytes=0;
+    uint64_t max_response_size_bytes=0;
+    std::vector<std::string> too_big_vars;
+
+    get_max_sizes_bytes(max_response_size_bytes, max_var_size_bytes, true);
+    BESDEBUG(MODULE, prolog << "max_var_size_bytes: " << max_var_size_bytes << "\n");
+    BESDEBUG(MODULE, prolog << "max_response_size_bytes: " << max_response_size_bytes << "\n");
+
+    auto response_size_bytes = compute_response_size_and_inv_big_vars(dds, max_var_size_bytes, too_big_vars);
```

Comment:
Lets merge this because the 'too big' requests are a problem that can crash the server. But, let's also profile this, after it is merged, to see if building the too_big_vars should be postponed until/if that info is needed.

Reply:
I have no idea what you mean, please explain.

Reply:
`1699007614|&|3363|&|timing|&|elapsed_us|&|23|&|start_us|&|1699007614778245|&|stop_us|&|1699007614778268|&|-|&|dap_utils::throw_if_too_big() - 
`